### PR TITLE
Further optimizations for cpu timing

### PR DESCRIPTION
**Pull request by ebrondol**
BEGINRELEASENOTES
- To optimize the timing of the algorithm, the setup of the LayerTiles is done only once, at the beginning of the event loop, when the algorithm is initialized, and not at the start of each event
- Layer tiles constants for CLIC have also been slightly optimized for a better performance
- A few methods have been added to printout the time report in case we want to test several repetitions on the same event - currently not used
- Time printout was included and the verbose printout of clue is now dependent on the OutputLevel variable set for the Gaudi algo
ENDRELEASENOTES

### CODE REVIEW

**Comment by ebrondol** [comment]
@jmcarcell any request for this PR or it can go in?

File: src/CLUEAlgo.cc [hunk 1]
```diff
@@ -192,6 +202,7 @@ void CLUEAlgo_T<TILES>::findAndAssignClusters(){
 
   // find cluster seeds and outlier
   std::vector<int> localStack;
+  localStack.reserve(10);
```
Comment (jmcarcell) [inline src/CLUEAlgo.cc:205]: Maybe instead of 10 some approximate fraction of `points_.n`?

File: src/ClueGaudiAlgorithmWrapper.cpp [hunk 2]
```diff
@@ -42,10 +42,72 @@ ClueGaudiAlgorithmWrapper::ClueGaudiAlgorithmWrapper(const std::string& name, IS
...
+  bool clue_verbose = false;
+  if (msgLevel(MSG::INFO) || msgLevel(MSG::DEBUG)){
+    clue_verbose = true;
+  }
+
+  auto start = std::chrono::high_resolution_clock::now();
+  clueAlgoBarrel_ = CLICdetBarrelCLUEAlgo(dc, rhoc, outlierDeltaFactor, clue_verbose);
+  auto finish = std::chrono::high_resolution_clock::now();
+  std::chrono::duration<double> elapsed = finish - start;
+  info() << "ClueGaudiAlgorithmWrapper: Set up time (Barrel): " << elapsed.count() * 1000 << " ms" << endmsg;
```
Comment (jmcarcell) [inline src/ClueGaudiAlgorithmWrapper.cpp:54]:
```suggestion
  info() << "ClueGaudiAlgorithmWrapper: Set up time (Barrel): " <<  std::chrono::duration_cast<std::chrono::milliseconds>(elapsed) << endmsg;
```
I'm not sure if it works but `std::cout << std::chrono::duration_cast<std::chrono::milliseconds>(elapsed) << std::endl;` and then you don't have to introduce the 1000 factor or the suffix (it's fine as it is but this is another way, also the cast can be done where you define `elapsed`

**Review by jmcarcell — APPROVED** [review]

**Comment by jmcarcell** [comment]
Whoops, sorry I forgot about this PR for a long time

File: src/CLUEAlgo.cc (hunk 1 above)
Comment (ebrondol) [reply src/CLUEAlgo.cc:205]: I am not sure that a fraction is a good way to go. We can introduce a better rule for the next round of changes.

**Pull request merged**